package/MDAnalysis/core/AtomGroup.py

```diff
@@ -1074,6 +1076,8 @@ def __add__(self, other):
         if not isinstance(other, (Atom, AtomGroup)):
             raise TypeError('Can only concatenate Atom or AtomGroup (not "{0}") to'
                             ' AtomGroup'.format(other.__class__.__name__))
+        if (self and other) and (not self.universe is other.universe):
```

Comment:
Then it is more readable if you check for the length of both atom groups.Or if they are empty.